pkg/node/getBalanceOf.go

```diff
@@ -0,0 +1,31 @@
+package node
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type GetBalanceOfResponse struct {
+	CandidateBalance float64
+	FinalBalance     float64
+}
+
+// GetBalanceOf returns as a float64 the candidate balance and final balance of an address
+func GetBalanceOf(client *Client, walletAddress string) (*GetBalanceOfResponse, error) {
+	addressDetails, err := Addresses(client, []string{walletAddress})
+	if err != nil {
+		return nil, fmt.Errorf("calling get_addresses :%w", err)
```

Comment:
If you look inside the code of Addresses, you will see that the reference to get_address is (almost, you could fix it ;-) ) already there. So this information is useless. Pass directly the error or wrap it smartly (by adding meaningful information).